conjure-error/src/error.rs

```diff
@@ -0,0 +1,380 @@
+use conjure_object::Value;
+use parking_lot::Mutex;
+use serde::Serialize;
+use std::borrow::Cow;
+use std::collections::hash_map::{self, HashMap};
+use std::error;
+use std::fmt;
+use std::ops::Index;
+use std::time::Duration;
+
+use crate::{ErrorType, Internal, SerializableError};
+
+/// Information about a throttle error.
+#[derive(Debug)]
+pub struct ThrottleError {
+    duration: Option<Duration>,
+}
+
+impl ThrottleError {
+    /// Returns the amount of time the client should wait before retrying, if provided.
+    #[inline]
+    pub fn duration(&self) -> Option<Duration> {
+        self.duration
+    }
+}
+
+/// Information about an unavailable error.
+#[derive(Debug)]
+pub struct UnavailableError(());
```

Comment:
Why is this a tuple struct instead of a unit struct? (i.e. `pub struct UnavailableError;`)

Reply:
We don't want to allow people to construct the type.